gallery_dl/extractor/subscribestar.py

```diff
@@ -84,44 +86,98 @@ def _login_impl(self, username, password):
-                    "name": text.unescape(text.extr(
-                        att, 'doc_preview-title">', '<')),
-                    "url" : text.unescape(text.extr(att, 'href="', '"')),
-                    "type": "attachment",
-                })
+        for key, config in media_types.items():
+            if key in media_config:
+                start, end, processor = media_config[key]
+                segments = (
+                    text.extract_all(
+                        html,
+                        ((key, start, end),),
+                    )[0],
+                )
+                for segment in segments:
+                    if segment[key]:
+                        content = processor(segment, key)
+                        if content:
+                            media.append(content)
+        return media
+
+    def _process_gallery_item(self, item, media_type):
+        gallery_list = util.json_loads(text.unescape(item["gallery"]))
+        for media in gallery_list:
+            if "/previews" in media["url"]:
+                self._warn_preview()
+            return {"url": media["url"], "type": media_type}
+
+    def _process_attachment_item(self, item, media_type):
+        return {
+            "id": text.parse_int(text.extr(item, 'data-upload-id="', '"')),
+            "name": text.unescape(text.extr(item, 'doc_preview-title">', "<")),
+            "url": text.unescape(text.extr(item, 'href="', '"')),
+            "type": media_type,
+        }
+
+    def _process_media_item(self, item, media_type):
+        if media_type == "link" and util.check_if_supported_by_ytdlp(
+                item[media_type]):
+            return {"url": "ytdl:" + item[media_type], "type": media_type}
+        elif media_type == "audio":
+            return {"url": item[media_type], "type": media_type}
 
+    def _media_from_post(self, html):
+        media_types = {
+            "gallery": True,
+            "attachments": True,
+            "link": True,
+            "audio": True,
+        }
+        media = self._extract_media(html, media_types)
         return media
 
     def _data_from_post(self, html):
         extr = text.extract_from(html)
+
+        links = (text.extract_all(html, (("url", 'data-href="', '"'),), )[0],)
+        audios = (text.extract_all(html, (("url", '<source src="',
+                                           '" type="audio/'),),)[0],)
+        gallery = text.extr(html, 'data-gallery="', '"')
+
+        content_type = None
+        if links and any(item["url"] for item in links):
+            content_type = "link"
+        if audios and any(item["url"] for item in audios):
+            content_type = "audio"
+        if gallery:
+            for item in util.json_loads(text.unescape(gallery)):
+                if item["type"] == "video":
+                    content_type = "video"
+                    break
+                else:
+                    content_type = "image"
+
```

Comment:
The goal of this part is to check what type of content the post contains to add a content_type to the post data so we can use it in directory name. This is not perfect as a post could probably contain multiple content_type, but I do not have enough samples to test.